### PR TITLE
[VAULT-33207] Update Policy Modal to add isHrefExternal to External Links

### PR DESCRIPTION
**Pull request by beagins**
### Description
What does this PR do? This is a followup PR to add `isHrefExternal` attribute to external HDS links in the Policy Modal
Original PR: https://github.com/hashicorp/vault/pull/29254

### Screenshots
ACL Policies: HCL Link
Screenshot 2024-12-26 at 3 24 58 PM

Endpoint Governing Policies: Sentinel Link
Screenshot 2024-12-26 at 3 28 52 PM

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this fix needs to be backported, use the appropriate `backport/` label that matches the desired release branch. Note that in the CE repo, the latest release branch will look like `backport/x.x.x`, but older release branches will be `backport/ent/x.x.x+ent`.
    - [ ] **LTS**: If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

**Comment by beagins** [comment]
`yarn test:ent`
```
1..219
# tests 219
# pass  209
# skip  10
# todo  0
# fail  0

# ok
```

File: ui/lib/core/addon/components/policy-example.hbs [hunk 1]
```diff
@@ -32,7 +32,10 @@
       Endpoint Governing Policies (EGPs) are tied to particular paths (e.g.
       <code class="tag is-marginless is-paddingless">aws/creds/</code>
       ) instead of tokens. They use
-      <Hds::Link::Inline @href="https://docs.hashicorp.com/sentinel/language">Sentinel</Hds::Link::Inline>
+      <Hds::Link::Inline
+        @isHrefExternal={{true}}
+        @href="https://docs.hashicorp.com/sentinel/language"
+      >Sentinel</Hds::Link::Inline>
       as a language to access
       <Hds::Link::Inline @href={{doc-link "/vault/docs/enterprise/sentinel/properties"}}>properties</Hds::Link::Inline>
```
Comment (hellobontempo) [inline ui/lib/core/addon/components/policy-example.hbs:40]: I beeline this needs one too. Then we're all set! "External" refers to any link outside the app so unless we're routing to another page inside the app, we should include this arg.

**Review by hellobontempo — CHANGES_REQUESTED** [review]

File: ui/lib/core/addon/components/policy-example.hbs (hunk 1 above)
Comment (beagins) [reply ui/lib/core/addon/components/policy-example.hbs:40]: Ah gotcha! I just applied it to the links that were previously using the `<ExternalLink/>` component, I'll update it. Thank you!!

**Review by hellobontempo — APPROVED** [review]
Thanks! ✨

File: ui/lib/core/addon/components/policy-example.hbs (hunk 1 above)
Comment (hellobontempo) [reply ui/lib/core/addon/components/policy-example.hbs]: That makes sense! The multiple use of "external" is confusing here. For additional context, we used to use this [DocLink](https://github.com/hashicorp/vault/blob/main/ui/lib/core/addon/components/doc-link.js#L22) component to render links to our documentation, which extended from [ExternalLink](https://github.com/hashicorp/vault/blob/main/ui/lib/core/addon/components/external-link.js#L23) and just provided the first part of the path. 

As we started using the HDS link components, we made the `doc-link` helper to use instead, which can be used directly in templates to build a doc link.

**Pull request merged**